types/node/dns.d.ts

```diff
@@ -31,9 +31,9 @@ declare module "dns" {
 
     // NOTE: This namespace provides design-time support for util.promisify. Exported members do not exist at runtime.
     namespace lookup {
-        function __promisify__(hostname: string, options: LookupAllOptions): Promise<{ address: LookupAddress[] }>;
-        function __promisify__(hostname: string, options?: LookupOneOptions | number): Promise<{ address: string, family: number }>;
-        function __promisify__(hostname: string, options?: LookupOptions | number): Promise<{ address: string | LookupAddress[], family?: number }>;
+        function __promisify__(hostname: string, options: LookupAllOptions): Promise<LookupAddress[]>;
+        function __promisify__(hostname: string, options?: LookupOneOptions | number): Promise<LookupAddress[]>;
```

Comment:
I'm quite sure that the `LookupOneOptions` and `LookupOptions ` variant is correct, see the last doc line for lookup at https://nodejs.org/dist/latest-v12.x/docs/api/dns.html#dns_dns_lookup_hostname_options_callback
Maybe the `| number` for the `LookupOptions` variant should be removed as this is better covered by the `LookupOneOptions` variant.

Reply:
I'm not following, please post the suggested diff.

Reply:
First variant as you suggested, second as it was before and third a combination of both:
```
function __promisify__(hostname: string, options: LookupAllOptions): Promise<LookupAddress[]>;
function __promisify__(hostname: string, options?: LookupOneOptions | number): Promise<{ address: string, family: number }>;
function __promisify__(hostname: string, options: LookupOptions ): Promise<{ address: string, family?: number } | LookupAddress[]>;
```